src/lucky/paginator/backend_helpers.cr

```diff
@@ -0,0 +1,91 @@
+module Lucky::Paginator::BackendHelpers
+  # Call this in your actions to paginate an query.
+  #
+  # This method will return a `Lucky::Paginator` object and the requested page
+  # of records.
+  #
+  # ## Examples
+  #
+  # ```crystal
+  # class Users::Index < BrowserAction
+  #   get "/users" do
+  #     # The 'UserQuery' will return just the records for the requested page
+  #     # because 'paginate' will add a 'limit' and 'offset' to the query.
+  #     pages, users = paginate(UserQuery.new)
+  #     render IndexPage, pages: pages, users: users
+  #   end
+  # end
+  #
+  # class Users::IndexPage < MainLayout
+  #   needs pages : Lucky::Paginator
+  #   needs users : UserQuery
+  #
+  #   def content
+  #     # Render 'users' like normal
+  #     mount Lucky::Paginator::SimpleNav.new(@pages)
+  #   end
+  # end
+  # ```
+  def paginate(
+    query : Avram::Queryable(T),
+    per_page : Int32 = paginator_per_page
+  ) : Tuple(Paginator, Avram::Queryable(T)) forall T
+    pages = Paginator.new \
+      page: paginator_page,
+      per_page: per_page,
+      item_count: query.clone.reset_order.reset_limit.reset_offset.select_count,
+      full_path: context.request.resource
```

Comment:
So with this, if you need to override what your page link paths are, you'd just pass in the string? `full_path: "/whatever"`

Reply:
In this case you can't since the method is hardcoded to using the current path. We could add a setter for `full_path` or an option on the `paginate` method, but I think this is a good start for MVP since I think 99% of the time you want to paginate the same url